### PR TITLE
feat: create ssh deployment target

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-29783]

help:
image

full interactive:
image

fully automated:
image

### CODE REVIEW

File: pkg/cmd/target/ssh/create/create.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+		},
+	}
+}
+
+func NewCmdCreate(f factory.Factory) *cobra.Command {
+	createFlags := NewCreateFlags()
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a ssh deployment target",
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/create/create.go:111]:
```suggestion
		Short: "Create a SSH deployment target",
```

File: pkg/cmd/target/ssh/create/create.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+		Long:  "Create a ssh deployment target in Octopus Deploy",
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/create/create.go:112]:
```suggestion
		Long:  "Create a SSH deployment target in Octopus Deploy",
```

File: pkg/cmd/target/ssh/create/create.go [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+		RunE: func(c *cobra.Command, _ []string) error {
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this Deployment Target.")
+	flags.StringVar(&createFlags.Account.Value, createFlags.Account.Name, "", "The name or id of the SSH Keypair or Username/Password account")
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/create/create.go:125]:
```suggestion
	flags.StringVar(&createFlags.Account.Value, createFlags.Account.Name, "", "The name or ID of the SSH Keypair or Username/Password account")
```

File: pkg/cmd/target/ssh/create/create.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+		`), constants.ExecutableName),
+		RunE: func(c *cobra.Command, _ []string) error {
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this Deployment Target.")
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/create/create.go:124]:
```suggestion
	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "A short, memorable, unique name for this deployment target.")
```

File: pkg/cmd/target/ssh/create/create.go [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+	if port == 0 {
+		port = 22
+	}
+	endpoint := machines.NewSSHEndpoint(opts.HostName.Value, port, opts.Fingerprint.Value)
+	endpoint.AccountID = account.GetID()
+
+	if opts.Runtime.Value == SelfContainedCalamari {
+		endpoint.DotNetCorePlatform = opts.Platform.Value
+	} else {
+		endpoint.DotNetCorePlatform = ""
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/create/create.go:170]: Is this necessary?

Comment (benPearce1) [reply pkg/cmd/target/ssh/create/create.go:170]: yes, if the user provides the mono runtime option, plus the platform

File: pkg/cmd/target/ssh/ssh.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	cmdCreate "github.com/OctopusDeploy/cli/pkg/cmd/target/ssh/create"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdSsh(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "ssh <command>",
+		Short:   "Manage ssh deployment targets",
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/ssh.go:15]:
```suggestion
		Short:   "Manage SSH deployment targets",
```

File: pkg/cmd/target/ssh/ssh.go [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+		Long:    "Work with Octopus Deploy ssh deployment targets.",
```
Comment (jbristowe) [inline pkg/cmd/target/ssh/ssh.go:16]:
```suggestion
		Long:    "Work with SSH deployment targets in Octopus Deploy.",
```

**Review by jbristowe — APPROVED** [review]

File: pkg/cmd/target/ssh/create/create.go (hunk 5 above)
Comment (jbristowe) [reply pkg/cmd/target/ssh/create/create.go:170]: I meant, isn't the default value an empty string?

Comment (benPearce1) [reply pkg/cmd/target/ssh/create/create.go:170]: Yes, but if both values are provided, should we take the runtime value as the source of truth, or is that a validation error?

Comment (benPearce1) [reply pkg/cmd/target/ssh/create/create.go:170]: correct, this is redundant

**Pull request merged**